ner_v2/detectors/numeral/number_range/README.md

```diff
@@ -0,0 +1,246 @@
+
+   2rd column corresponds to the position of the keyword. If 1 means it will be present after number value, -1 if present before number.  
+
+   3rd column defines the range type of number i.e if it contains min value or max value or both.    
+
+   **For example** - *mujhe 2000 se jada log chahiye kal ki rally me* . In this text, range keyword `se jada` which is present in position `1` (after number) defines that number `2000` is the minimum value defined in the text, since no maximum value specified in text, hence it will be null.
+
+
+#### Guidelines to add new detectors for number range apart from builtin ones:
+
+Create a new class `number_range_detection.py`  inside language directory.
+To start with copy the code below
+
+```python
+import re
+import os
+
+from ner_v2.constant import LANGUAGE_DATA_DIRECTORY
+from ner_v2.detectors.numeral.constant import NUMBER_REPLACE_TEXT
+from ner_v2.detectors.numeral.number_range.standard_number_range_detector import BaseNumberRangeDetector
+
+
+class NumberRangeDetector(BaseNumberRangeDetector):
+    data_directory_path = os.path.join((os.path.dirname(os.path.abspath(__file__)).rstrip(os.sep)),
+                                       LANGUAGE_DATA_DIRECTORY)
+
+    def __init__(self, entity_name, language, unit_type=None):
+        super(NumberRangeDetector, self).__init__(entity_name=entity_name,
+                                                  language=language,
+                                                  data_directory_path=NumberRangeDetector.data_directory_path,
+                                                  unit_type=unit_type)
+```
+
+Note that the class name must be `NumberRangeDetector` 
+and should inherit from `ner_v2.detectors.numeral.number_range.standard_number_range_detector.BaseNumberRangeDetector`
+
+Next, we define a custom detector. For our purposes, we will add a detector to detect number range from text of patter 'between *number1* and *number2*. It will detect number1 and number2 as minimum value and maximum value respectively.
+
+1. The custom detector must accept two arguments `number_range_list` and `original_list` and must operate on `self.processed_text (This text contained tagged number text i.e number will be replaced with __dnumber__<number>`). So in your pattern, you just have to include pattern with number tag instead of `\d+` for the number. (as it handle all number having decimal, ordinal and integer)
+2. The two arguments `number_range_list` and `original_list` can both be either None or lists of the same length.
+   `number_range_list` contains parsed min and max value  along with unit and `original_list` contains their corresponding text substrings in the passed text that were detected as numbers range.
+3. Your detector must append parsed number range dicts with keys `'min_value', 'max_value', 'unit'` to `number_range_list`
```

Comment:
This point does not apply here.

```
3. Unlike other detectors, once you detect min part and/or max part of a range you must call
`ner_v2.detectors.numeral.number_range.standard_number_range_detection.BaseNumberDetector. _get_number_range` i.e. `self. _get_number_range` which takes three str/unicode arguments 
- `min_part_match` - number part of the `self.processed_text` that was detected as a minimum part of the range. E.g.  `from __number__0 to  __number__1`. In this case `min_part_match='__number__0'`. This can be None if no min part was detected.
-  `max_part_match` - number part of the `self.processed_text` that was detected as a minimum part of the range. E.g. `from __number__0 to  __number__1`. In this case `max_part_match='__number__1'`. This can be None if no max part was detected.
-  `full_match` - the entire part of the `self.processed_text` that denotes a range, a substring that contains both `min_part_match` and `max_part_match` as its substrings.
E.g. `from __number__0 to  __number__1`. In this case `full_match='from __number__0 to  __number__1'`. This can be None if no max part was detected.
```